### ISSUE TITLE
Support for Geofences?

### ISSUE THREAD

**Issue by maurizi** [mapzen/LOST#10]
Is there any intention to add support for Geofences and the Geofence API?

**Comment by friesenkiwi** [comment]
I am currently trying to port OwnTracks from PlayServices to LOST (https://github.com/owntracks/android/issues/88). It looks good so far, but missing Geofencing would be a dealbreaker, since it is a feature many users of it heavily rely upon.
For the current development state, I added some try/catch for the "not yet implemented" RuntimeExceptions to at least not crash the app.
But we would still like to switch and thus "need" GeoFencing support in LOST. Are there any recent considerations of implementation or forseeable progress?
Thinking about it, it might be not too complicated to add it? As far as I can see, it would be implementing the three methods addGeofences(List<Geofence>, PendingIntent), removeGeofences(List<String>), removeGeofences(PendingIntent) in GeofencingApiImpl.java
and implement the Geofence.java interface, is that correct? Also, of course, the actual detection of transitions needs to be implemented. But since there are only circular Fences, that should be quite straightforward, right?
I found https://code.google.com/p/openmap/source/browse/src/openmap/com/bbn/openmap/geo/Intersection.java
or
http://www.codeproject.com/Articles/62482/A-Simple-Geo-Fencing-Using-Polygon-Method
as a start, but they seem to be eben a little bit too complex.
And I'd say, a minimal set of features would already be desireable, no need to implement all features at once? (Loitering, Dwelling, Responsiveness, GeofencingRequest etc.)

**Comment by xbroak** [comment]
Hows this going? Any sign of addProximityAlert methods?

**Comment by friesenkiwi** [comment]
Hey, there. Sorry I had to drop my first work on this last year, but I am right now looking into it again. I see @ecgreb you are working on it as well? How far is the progress? Would it be any helpful if I joined in?
Is the schedule for 2.0 very tight? Because I would love to see this feature in a release soon, so it can be used over at owntracks, and I would be very happy to help, if that's of interest.

**Comment by ecgreb** [comment]
Hi @friesenkiwi. Its all good. Help is always appreciated. Here is what I have so far:

https://github.com/mapzen/LOST/pull/86

As you can see in the description things are not quite working yet. Also once inital support is in place there will be more work needed including registering multiple geofences, removing geofences, and triggering geofences via mock locations. Plus likely some other things I am overlooking.

**Comment by friesenkiwi** [comment]
Hi! Awesome work, thank you for your code!
I added some adding and removing support and also merged the current master and bumped the build environment a bit (without the additional dcendents dependency it wasn't building on my machine).
What do you say? I hope, the way to implement managing / adding / removing Geofences by HashMap is acceptable?
Also, I'm not yet used to using checkstyle. I installed the plugin in Android Studio, but it only shows "The default Checkstyle rules" in the scan pane. How can I use the xmls from the repo?

**Comment by ecgreb** [comment]
Closing in favor of [this list of issues](https://github.com/mapzen/LOST/pull/91#issuecomment-245731940) created for outstanding parts of the geofencing API that still need to be implemented.
